glue_vispy_viewers/volume/volume_toolbar.py

```diff
@@ -9,27 +9,83 @@
+        :param event:
+        """
         self.selection_origin = event.pos
+        print('event.pos', self.selection_origin)
         if self.mode is 'point':
-            # TODO: add dendrogram selection here
-            pass
+            pos = self.get_ray_line()
 
-    def on_mouse_release(self, event):
+            visible_data, visual = self.get_visible_data()
+            data_object = visible_data[0]
+            data_array = data_object['PRIMARY']
+
+            trans = self._vispy_widget.limit_transforms[visual].translate
+            scale = self._vispy_widget.limit_transforms[visual].scale
+            scaled_trans = (trans/scale).tolist()
+
+            # find the intersected points position in visual coordinate through z axis
+            inter_points_index = np.zeros((data_array.shape[0], 3))
+
+            for z in range(-data_array.shape[0]/2, data_array.shape[0]/2):
+                #   3D line defined with two points (x0, y0, z0) and (x1, y1, z1) as
+                #   (x - x1)/(x2 - x1) = (y - y1)/(y2 - y1) = (z - z1)/(z2 - z1) = t
+                t = (z - pos[0][2])/(pos[1][2] - pos[0][2])
+                inter_points_index[z, 0] = x = t * (pos[1][0] - pos[0][0]) + pos[0][0]
+                inter_points_index[z, 1] = y = t * (pos[1][1] - pos[0][1]) + pos[0][1]
+                inter_points_index[z, 2] = z
+
+            # cut the line within the cube
+            inter_points_index = inter_points_index[inter_points_index[:, 0] > -data_array.shape[2]/2]
+            inter_points_index = inter_points_index[inter_points_index[:, 0] < data_array.shape[2]/2]
+            inter_points_index = inter_points_index[inter_points_index[:, 1] > -data_array.shape[1]/2]
+            inter_points_index = inter_points_index[inter_points_index[:, 1] < data_array.shape[1]/2]
+
+            # get the intersected points index with the same shape as data_array
+            inter_points_index = inter_points_index.astype(int)-scaled_trans[:3]
+
+            # set colors for markers
+            colors = np.ones((inter_points_index.shape[0], 4))
+            colors[:] = (0.5, 0.5, 0, 1)
+
+            inter_value = []  # value of intersected points
+            for each_pos in inter_points_index:
+                each_pos = tuple(each_pos[::-1])   # order of data_array is [z, y, x]
+                # here data_array is different from self.vol_data, it's a Glue Data object
+                inter_value.append(data_array[each_pos])
+
+            max_data_value = inter_points_index[np.argmax(inter_value), :3]
+            colors[np.argmax(inter_value)] = (1, 1, 0.5, 1)  # change the color of max value marker
+
+            self.markers.set_data(pos=inter_points_index, face_color=colors)
+            print('interpos is', inter_points_index, data_array)
+            # update limit to make transform sync with the stretch  :)
+            self._vispy_widget.add_data_visual(self.markers)
+            self._vispy_widget._update_limits()
```

Comment:
Here is the line that I tried to set the transform of `Markers/ray line` to the normalized which makes it sync when doing stretch on UI control. But it doesn't work as I expected :(